projects/packages/partner/tests/php/test-partner-coupon.php

```diff
@@ -229,4 +232,94 @@ public function dataprovider_purge_dates() {
 		);
 	}
 
+	/**
+	 * Purge coupon: verify remote availability check is respected.
+	 *
+	 * @dataProvider dataprovider_availability_check_scenarios
+	 *
+	 * @param array $mock_response Data used to mock response.
+	 * @param bool  $expectation The expected assertion result.
+	 */
+	public function test_maybe_purge_coupon_by_availability_check( $mock_response, $expectation ) {
+		$this->setup_coupon();
+		$coupon_code = sprintf( '%s_%s_%s', self::PARTNER['prefix'], self::PRESET['code'], 'abc123' );
+
+		Jetpack_Options::update_options(
+			array(
+				Partner_Coupon::$coupon_option => $coupon_code,
+				Partner_Coupon::$added_option  => time(),
+				'id'                           => 123,
+			)
+		);
+
+		if ( isset( $mock_response['body'] ) ) {
+			$mock_response['body'] = wp_json_encode( $mock_response['body'] );
+		}
+
+		$mock_client = $this->getMockBuilder( \stdClass::class )
+							->setMethods( array( 'wpcom_json_api_request_as_blog' ) )
+							->getMock();
+
+		$mock_client
+			->expects( $this->once() )
+			->method( 'wpcom_json_api_request_as_blog' )
+			->willReturn( $mock_response );
```

Comment:
We can use an anonymous function instead of dealing with the mock builder:
```
$mock_requester = function () use ( $mock_response ) { return $mock_response; };
$instance = new Partner_Coupon( $mock_requester );
```

Reply:
It's is surely much simpler to read!
I would prefer to keep the `expects( $this->once() )` check since we make a remote request that will block the page from loading until it's done. If we - for any reason - make the request more than once, then it's 1) an error and 2) a bad user experience.

Although the above could be argued to belong in its own test, but lets keep it simple 😄